src/rule_tester.rs

```diff
@@ -31,55 +37,55 @@ impl RuleTester {
     }
 
     fn run_valid_test(&self, valid_test: &RuleTestValid) {
-        let tmp_dir = TempDir::new("valid_test").unwrap();
-        let test_filename = "tmp.rs";
-        fs::write(tmp_dir.path().join(test_filename), &valid_test.code).unwrap();
-        Command::cargo_bin("tree-sitter-lint")
-            .unwrap()
-            .args(["--rule", &self.rule.name])
-            .current_dir(tmp_dir.path())
-            .assert()
-            .success()
-            .stdout(predicate::str::is_empty());
+        let violations = crate::run_for_slice(
+            valid_test.code.as_bytes(),
+            "tmp.rs",
```

Comment:
This seems to imply that there will need to be a way to generate "dummy filenames" per `SupportedLanguage`?

Reply:
Ah yeah